etcdmain/etcd.go

```diff
@@ -279,6 +279,10 @@ func startProxy(cfg *config) error {
 		if err != nil {
 			return err
 		}
+		// os.MkdirAll creates directory before umask so make sure that a desired permission is set
+		if err = fileutil.IsDirWriteable(cfg.ec.Dir); err != nil {
```

Comment:
`err = os.MkdirAll(cfg.ec.Dir, fileutil.PrivateDirMode)` means that `cfg.ec.Dir` must be writeable for the etcd owner.
Is this code redundant?

Reply:
@cfc4n thanks, the when a new dir is created by MkdirAll. as documented https://golang.org/pkg/os/#MkdirAll , it may not respect the the filer mode used (in our case 700) based on umask setting on the system by admin. So this is check is to make sure that the directory is created as expected by the MkdirAll. Hope that answers your question. Thanks!

Reply:
Thanks, I see.